### ISSUE TITLE
Config files copied by Dreamview

### ISSUE THREAD

**Issue by arturvasilev** [ApolloAuto/apollo#13097]
When selecting car in Dreamview -- many files in repo copied from `modules/calibration/data` into module's folders. It is not necessary and make `git status` dirty.

Here is result of `git status` after running `bootstrap.sh` and selecting vehicle `Mkz Example`:
```git
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   modules/common/data/vehicle_param.pb.txt
	modified:   modules/control/conf/navigation_lincoln.pb.txt
	modified:   modules/drivers/gnss/conf/gnss_conf.pb.txt
	modified:   modules/drivers/lidar/velodyne/conf/velodyne128_conf.pb.txt
	modified:   modules/drivers/lidar/velodyne/conf/velodyne16_front_center_conf.pb.txt
	modified:   modules/drivers/lidar/velodyne/conf/velodyne16_rear_left_conf.pb.txt
	modified:   modules/drivers/lidar/velodyne/conf/velodyne16_rear_right_conf.pb.txt
	modified:   modules/drivers/lidar/velodyne/conf/velodyne16_up_center_conf.pb.txt
	modified:   modules/drivers/lidar/velodyne/conf/velodyne64_conf.pb.txt
	modified:   modules/drivers/lidar/velodyne/params/64E_S3_calibration_example.yaml
	modified:   modules/drivers/lidar/velodyne/params/velodyne64_novatel_extrinsics_example.yaml
	modified:   modules/drivers/video/conf/video_front_12mm.pb.txt
	modified:   modules/drivers/video/conf/video_front_6mm.pb.txt
	modified:   modules/drivers/video/conf/video_left_fisheye.pb.txt
	modified:   modules/drivers/video/conf/video_rear_6mm.pb.txt
	modified:   modules/drivers/video/conf/video_right_fisheye.pb.txt
	modified:   modules/localization/msf/params/gnss_params/ant_imu_leverarm.yaml
	modified:   modules/localization/msf/params/velodyne_params/velodyne64_height.yaml
	modified:   modules/localization/msf/params/velodyne_params/velodyne64_novatel_extrinsics_example.yaml

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	modules/drivers/lidar/velodyne/params/velodyne64_height.yaml
	modules/perception/data/params/64E_S3_calibration_example.yaml
	modules/perception/data/params/front_camera_extrinsics.yaml
	modules/perception/data/params/front_camera_intrinsics.yaml
	modules/perception/data/params/long_camera_extrinsics.yaml
	modules/perception/data/params/long_camera_intrinsics.yaml
	modules/perception/data/params/radar_extrinsics.yaml
	modules/perception/data/params/radar_front_extrinsics.yaml
	modules/perception/data/params/short_camera_extrinsics.yaml
	modules/perception/data/params/short_camera_intrinsics.yaml
	modules/perception/data/params/velodyne64_height.yaml
	modules/perception/data/params/velodyne64_novatel_extrinsics_example.yaml
```

There are ready-to-use functionality in dreamview when it uses `.dag` files from `modules/calibration/data` of concrete car. Without copying any files.

Is there any reason to copy these files into module folder?
My current hypothesis: someone was lazy for a minute, then it grew up into crutch and bad practice.
My advice (if my hypothesis is correct): Delete crutch in Dreamview backend  which copies files ; fix `.dag`and config-files appropriately, so all of them will lie in `modules/calibration/data/<concrete-vehicle>`.

### System information

- Linux Ubuntu 20.04
- Apollo installed from source
- Apollo version 6.0
- Output of `apollo.sh config` on `master` branch:
```
[avasilev@in-dev-docker:/apollo]$ ./apollo.sh config
[INFO] Apollo Environment Settings:
[INFO]     APOLLO_ROOT_DIR: /apollo
[INFO]     APOLLO_CACHE_DIR: /apollo/.cache
[INFO]     APOLLO_IN_DOCKER: true
[INFO]     APOLLO_VERSION: master-2020-11-30-2ef47f1143
[INFO]     DOCKER_IMG: dev-x86_64-18.04-20201110_0617
[INFO]     APOLLO_ENV:  STAGE=dev USE_ESD_CAN=false
[INFO]     USE_GPU: USE_GPU_HOST=1 USE_GPU_TARGET=1
[INFO] Usage: /apollo/scripts/apollo_config.sh [Options]
[INFO] Options:
[INFO]     -i|--interactive      Run in interactive mode
[INFO]     -n|--noninteractive   Run in non-interactive mode
[INFO]     -h|--help             Show this message and exit
```

### Steps to reproduce the issue:
- Start container with `./dev_start.sh` on GPU-capable host
- Build in container with `./apollo.sh build_gpu`.
- Launch Dreamview with `bootstrap.sh`.
- Select vehicle `Mkz Example` or any other.

**Comment by daohu527** [comment]
Your suggestion is good, and I like to discuss more.

1. If only one configuration file is used, it must always be modified to fit multiple vehicles. Regardless of whether you modify the config or the path in the dag file. 
2. If you use multiple configuration files or dag, you can select files without modifying any files. 

so what's the best practice?

**Comment by arturvasilev** [comment]
## Separate config files

Best practice would be to separate parts of modules config -- which is car-related or not. Car related part of config file should lays in `modules/calibration/data/<concrete-car>`. And another part, which is common for _any_ possible car, should stay in module's folder.

For example -- for velodyne driver we can store timings, azimuth of lasers and so on in module's folder: `modules/drivers/velodyne/conf/...`. Because thay are factory-defined and we can just get them from manual/spec. But IP:port and name of the channel for read/write are car-related, so it should be in `modules/calibration/data`.

In my practice there are few modules which has to have non-car-related files. Almost always we can hardcode such common-info into module itself, and consume single config, which is car-related.

## Select vehicle from dreamview

These vehicles are defined [here](https://github.com/ApolloAuto/apollo/tree/master/modules/dreamview/conf/hmi_modes). Now they points to `.dag`-files in module's folders. We should change that. They should point to `.dag`-files in `modules/calibration/data/<concrete-car>`.

In `modules/calibration/data` already exist [convention](https://github.com/ApolloAuto/apollo/tree/master/modules/calibration/data) how to organize files there.

**Comment by daohu527** [comment]
Therefore, we use multiple dag files instead of copying and replacing files ? Most case is suitable but the yaml file, so maybe dag should support yaml file config path.

**Comment by arturvasilev** [comment]
Yes! 

We can set path to `.yaml` file via gflags or in string param inside `.pb.txt`. So we can store them in calibration/data` too.  There is no need to add support `.yaml`-paths in component constructor.

If some modules consume yaml-config — maybe we should rewrite them so they accept protobuf?

**Comment by arturvasilev** [comment]
Any thoughts? Or it has low priority?

**Comment by Robotics010** [comment]
agree, it prevents me from finding all the files for the next commit, because there are a lot of changes in repo.
